Source/WebCore/css/SelectorChecker.cpp

```diff
@@ -1223,7 +1223,22 @@ bool SelectorChecker::checkOne(CheckingContext& checkingContext, const LocalCont
                 return true;
             if (checkingContext.pseudoId != PseudoId::Highlight || !selector.argumentList())
                 return false;
-            return selector.argumentList()->first() == checkingContext.nameForHightlightPseudoElement;
+            return selector.argumentList()->first() == checkingContext.nameIdentifier;
+
+        case CSSSelector::PseudoElementViewTransitionGroup:
+        case CSSSelector::PseudoElementViewTransitionImagePair:
+        case CSSSelector::PseudoElementViewTransitionOld:
+        case CSSSelector::PseudoElementViewTransitionNew: {
+            // Always matches when not specifically requested so it gets added to the pseudoIdSet.
+            if (checkingContext.pseudoId == PseudoId::None)
+                return true;
```

Comment:
Can you explain how this works? I see the existing highlight pseudo support does the same. This won't make `element.matches("::view-transition-old(*)")` return true will it?

Reply:
> Can you explain how this works? I see the existing highlight pseudo support does the same.

It seems to activate this codepath which matches what the comment says: https://searchfox.org/wubkat/rev/cf345a7967b324872fb7ff4a173307ddb93101ac/Source/WebCore/css/SelectorChecker.cpp#295-298

> This won't make element.matches("::view-transition-old(*)") return true will it?

From my testing, it doesn't.